uwblueprint/jump-math
# (3) feat: view students in class

## Notion ticket link

[View Students in Class](https://www.notion.so/uwblueprintexecs/View-Students-in-Class-94167c082d3e43c898f3377228259615)

_Reviewing by commit might make the PR easier to review!_



## Implementation description
- [Refactor directory structure to be clearer](https://github.com/uwblueprint/jump-math/pull/424/commits/decb209b617c0dcdaaf8e96b2e951c77c46d443b)
- [Add basic table with no sorting/searching](https://github.com/uwblueprint/jump-math/pull/424/commits/37bd02c216af75f281d7e72ef8b288cbb1e7c9b1)
- [Extract generic filterable/searchable page component](https://github.com/uwblueprint/jump-math/pull/424/commits/8e492214179eb54a169154f8533f421778084287)
- [fix: Allow student response type to be indexed](https://github.com/uwblueprint/jump-math/pull/424/commits/6710b18e3e372ee5295415486b876ca71dd44b2f)
- [Make student table searchable/sortable](https://github.com/uwblueprint/jump-math/pull/424/commits/16904af98a8c6a72dae296e6163c4d095807f38a)
- [Add more mock data](https://github.com/uwblueprint/jump-math/pull/424/commits/60e53480e1113f94b17707cec7786e5029c8422b)
- [Add in error and loading states](https://github.com/uwblueprint/jump-math/pull/424/commits/7255f26304bc219db174d4de16d37be154b000c0)
- [Move "Mobile Redirect" illustration](https://github.com/uwblueprint/jump-math/pull/424/commits/d93e1df9a44706f7dd3bfd7019f3b4f1ff6a7307) 
- [Refactor RouterTabs component to support element prop](https://github.com/uwblueprint/jump-math/pull/424/commits/bc90bb90b77f3861e4b161a72b59bc22b115ce42)
- [Update error message dynamically depending on the page](https://github.com/uwblueprint/jump-math/pull/424/commits/edeebed0681f4144fbaef6995506fec58a1f54b8)
- [fix: RouterTabs component shouldn't double-push routes](https://github.com/uwblueprint/jump-math/pull/424/commits/1941f2eab149fe1929eddf2e4bb1169891e30bd8)
- [chore: extract RedirectTo component to redirect to a paramified URL](https://github.com/uwblueprint/jump-math/pull/424/commits/0baedd1c0847bd9dfd218c7cab68ab5e4b15cd4f)

<img width="1416" alt="image" src="https://github.com/uwblueprint/jump-math/assets/9922514/6a197624-e651-4b6b-89ad-c39172892860">




## Steps to test
1. View the class list page.
2. Click on a class card.
3. Verify you are redirected to a drilldown page with a populated, searchable, and sortable student list.



## What should reviewers focus on?
* 


## Checklist
